physical/raft/fsm.go

```diff
@@ -633,6 +637,116 @@ func listPageInner(ctx context.Context, tx *bolt.Tx, prefix string, after string
 	return keys, nil
 }
 
+func (f *FSM) applyBatchNonTxOps(b *bolt.Bucket, txnState *fsmTxnCommitIndexApplicationState, command *LogData) error {
+	for _, op := range command.Operations {
+		var err error
+		switch op.OpType {
+		case putOp:
+			err = b.Put([]byte(op.Key), op.Value)
+			if err == nil {
```

Comment:
Add comment: this is a direct write so it should only be tracked if the write itself succeeded.